### PR TITLE
feat: improve proposal details visualization

### PR DESCRIPTION
**Pull request by manu0466**
## Description

Closes: DPM-162, DPM-163, DPM-172, DPM-178

This PR enhances the visualization of the details of a governance proposal with the following changes:

1. Displays the proposal messages inside the scrollview instead of a horizontal list.
2. Generalizes the visualization of the modules `MsgUpdateParams` to display the changed parameters.
3. Adds the visualization of the proposal summary.
4. Shows the profile of the proposal's proposer.

###Final result:
output

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] provided a link to the relevant issue or specification
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed

### CODE REVIEW

File: src/screens/GovernanceProposalDetails/components/ProposalDetails/index.tsx [hunk 1]
```diff
@@ -79,6 +65,15 @@ const ProposalDetails: React.FC<ProposalDetailsProps> = ({ proposal }) => {
   return (
     <View>
       <Spacer paddingTop={24} />
+      {/* Summary */}
```
Comment (manu0466) [inline src/screens/GovernanceProposalDetails/components/ProposalDetails/index.tsx:68]: @RiccardoM I display the proposal summary above the description. Let me know if this is fine.

Comment (RiccardoM) [reply src/screens/GovernanceProposalDetails/components/ProposalDetails/index.tsx:67]: Yea it's perfect :+1:

File: src/components/Messages/common/MsgUpdateParamsDetails/index.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+      if (typeof objectValue === 'object') {
+        // Special case for the coin object.
+        if (typeof objectValue.denom === 'string' && typeof objectValue.amount === 'string') {
+          serializedValue = formatCoin(objectValue);
+        } else {
+          serializedValue = JSON.stringify(objectValue, undefined, 4);
+        }
+      } else {
+        serializedValue = objectValue.toString();
```
Comment (RiccardoM) [inline src/components/Messages/common/MsgUpdateParamsDetails/index.tsx:32]: Does this work on boolean values as well? I saw in your preview Gif that the `send_enabled` parameter was rendered as a white square box for some reason. 

Also, would it be better to check if the `objectValue` is either `undefined` or `null` and render it as `null` in both cases?

**Review by RiccardoM — COMMENTED** [review]
Looks good to me. Just a minor comment on the rendering of the parameter values

File: src/components/Messages/common/MsgUpdateParamsDetails/index.tsx (hunk 2 above)
Comment (manu0466) [reply src/components/Messages/common/MsgUpdateParamsDetails/index.tsx:40]: @RiccardoM I have handled the case where object value is `undefined` or `null`.
About the booleans there was an error in the conversion of the `/desmos.tokenfactory.v1.MsgUpdateParams`.
I have update it to matche the [proto definition](https://github.com/desmos-labs/desmos/blob/83c3da9d96f35828c2e9b72ead2e6053941e37a9/proto/desmos/tokenfactory/v1/params.proto#L13).

**Review by RiccardoM — APPROVED** [review]

**Pull request merged**